utils/postDailyLeetcodeMessage.js

```diff
@@ -0,0 +1,125 @@
+    return;
+  }
+
+  const dailyProblemData =
+    dailyLeetcodeData.data.activeDailyCodingChallengeQuestion;
+
+  const weeklyLeetcodeData = await weeklyGetLeetcodeData();
+  if (!weeklyLeetcodeData) {
+    console.error("there's no data available");
+    return;
+  }
+
+  const weeklyProblemData =
+    weeklyLeetcodeData.data.dailyCodingChallengeV2.weeklyChallenges;
+  const lastWeeklyProblemData = weeklyProblemData[weeklyProblemData.length - 1];
+
+  const oneDay = 24 * 60 * 60 * 1000; //this is a day expressed in milliseconds
+  const now = new Date();
+  const weeklyChangeDate = Date.parse(lastWeeklyProblemData.date) + oneDay * 7;
+  const weeklyRemainingDays = Math.round(
+    Math.abs((weeklyChangeDate - now) / oneDay)
+  );
+  const weeklyRemainingDaysMessage =
+    weeklyRemainingDays + (weeklyRemainingDays >= 2 ? ' days' : ' day');
+
+  return {
+    weeklyRemainingDaysMessage,
+    dailyProblemData,
+    lastWeeklyProblemData,
+  };
+};
+
+export async function generateDailyLeetcodeMessage() {
+  // Find channel
+  const {
+    dailyProblemData,
+    lastWeeklyProblemData,
+    weeklyRemainingDaysMessage,
+  } = await fetchLeetCodeData();
+
+  const dailyProblemMessage = new MessageEmbed()
+    .setColor('#00FFFF')
+    .setTitle(
+      `${dailyProblemData.question.frontendQuestionId}. ${dailyProblemData.question.title}`
+    )
+    .setURL(`${LEETCODE_URL}${dailyProblemData.link}`)
+    .addFields(
+      {
+        name: 'Difficulty',
+        value: '```' + dailyProblemData.question.difficulty + '\n```',
+        inline: true,
+      },
+      {
+        name: 'Success rate',
+        value:
+          '```' +
+          Number.parseFloat(dailyProblemData.question.acRate).toFixed(2) +
+          '```',
+        inline: true,
+      }
+    );
+
+  const weeklyProblemMessage = new MessageEmbed()
+    .setColor('#FFBF00')
+    .setTitle(
+      `${lastWeeklyProblemData.question.questionFrontendId}. ${lastWeeklyProblemData.question.title}`
+    )
+    .setURL(`${LEETCODE_URL}${lastWeeklyProblemData.link}`)
+    .addFields({
+      name: 'Remaining time',
+      value: `${weeklyRemainingDaysMessage}`,
+      inline: false,
+    })
+    .setFooter({ text: 'Time to code 🔥👨‍💻🔥' });
+
+  return { weeklyProblemMessage, dailyProblemMessage };
+}
+
+export async function postDailyLeetcodeMessage() {
+  const { weeklyProblemMessage, dailyProblemMessage } =
+    await generateDailyLeetcodeMessage();
+
+  const channelName = config.LEETCODE_CHALLENGES_CHANNEL || undefined;
+
+  const channel = discordClient.channels.cache.find(
+    (channel) => channel.name === channelName
+  );
+
+  if (!channel) {
+    console.error(`could not find Discord channel: ${channelName}`);
+    return;
+  }
+
+  await channel.send({
+    content: '**Leetcode Daily**',
+    embeds: [dailyProblemMessage],
+  });
+
+  await channel.send({
+    content: '**Leetcode Weekly**',
+    embeds: [weeklyProblemMessage],
+  });
+}
```

Comment:
The formatting in this file is different than all the other files. I suggest using the same formatting we had or change it everywhere and explain what we should do so we don't change the formatting when pushing code.